exp/api/v1alpha4/azuremanagedmachinepool_types.go

```diff
@@ -21,8 +21,21 @@ import (
 	capierrors "sigs.k8s.io/cluster-api/errors"
 )
 
+const (
+	// LabelAgentPoolMode represents mode of an agent pool. Possible values include: System, User.
+	LabelAgentPoolMode = "azurecluster.infrastructure.cluster.x-k8s.io/agentpoolmode"
+
+	// SystemNodePool represents mode system for azuremachinepool.
+	SystemNodePool = "System"
```

Comment:
nit: you can probably just use the SDK values AgentPoolMode{System,User}, or else I'd prefer naming slightly closer to that (SystemNodePool is a bit too generic imo and could easily be a var name used elsewhere)

Reply:
actually I think keeping the label definition here is better..

maybe Label{System,User}NodePool? not super picky on this one tbh

Reply:
ack 👍

Reply:
Will this be fine ? 

```
type NodePool string
const (
	// AgentPoolModeSystem represents mode system for azuremachinepool.
	AgentPoolModeSystem NodePool = "System"
)
```

Reply:
I’d probably 

```go
type NodePoolMode string
const (
  NodePoolModeSystem …
  NodePoolModeUser…
)
```

or something to align the names a bit more closely